java/code/src/com/suse/manager/webui/controllers/StatesAPI.java

```diff
@@ -159,6 +134,104 @@ public static String stateContent(Request request, Response response, User user)
+     *
+     * @param request the request object
+     * @param response the response object
+     * @param user the authorized user
+     * @return the result schedule object
+     */
+    public static String singleSchedule(Request request, Response response, User user) {
+        String scheduleId = request.params("scheduleId");
+        Map<String, String> schedule = getSingleSchedule(scheduleId, user);
+        if (schedule.isEmpty()) {
+            return json(response,
+                    ResultJson.error("Schedule not found"));
+        }
+        return json(response,
+                ResultJson.success(schedule));
+    }
+
+    /**
+     * Creates a list of objects from a list of {@link TaskoSchedule} instances
+     *
+     * @param user the authorized user
+     * @return the list of Recurring State Apply Schedules
+     */
+    private static List<Map<String, String>> getSchedules(User user) {
+        /* TODO: Check user accessability */
+        try {
+            List<Map> taskoSchedules = new TaskomaticApi().findActiveSchedulesByBunch(user,
+                    "recurring-state-apply-bunch");
+            return taskoSchedules.stream().map(taskoSchedule -> {
+                Map<String, String> schedule = new HashMap<>();
+                Map<String, String> data = (Map<String, String>) taskoSchedule.get("data_map");
+                schedule.put("targetType", data.get("targetType"));
+                schedule.put("minionIds", data.get("minionIds"));
+                schedule.put("isActive", data.get("isActive"));
+                schedule.put("scheduleId", taskoSchedule.get("id").toString());
+                schedule.put("scheduleName", taskoSchedule.get("job_label").toString());
+                String date = new Timestamp(((Date) taskoSchedule.get("active_from")).getTime()).toString();
+                schedule.put("createdAt", date.substring(0, date.indexOf(".")));
+                schedule.put("frequency", taskoSchedule.get("cron_expr").toString());
+                return schedule;
+            }).collect(Collectors.toList());
+        }
+        catch (TaskomaticApiException e) {
+            return new ArrayList<>();
+        }
+    }
+
+    /**
+     * Returns a single schedule with the given scheduleId
+     *
+     * @param scheduleId the id of the schedule
+     * @param user the authorized user
+     * @return the result schedule object
+     */
+    private static Map<String, String> getSingleSchedule(String scheduleId, User user) {
+        try {
+            Map<String, String> schedule = new HashMap<>();
+            Map taskoSchedule = new TaskomaticApi().findScheduleById(user, Long.parseLong(scheduleId));
+            if (taskoSchedule.get("id") != null) {
+                String cronExpr = taskoSchedule.get("cron_expr").toString();
+                schedule = (Map<String, String>) taskoSchedule.get("data_map");
+                if (schedule.get("minute").isEmpty()) {
```

Comment:
`minute` is the only one that gets set regardless of which schedule type the user selected, with the exception of a `cron` String parsed by the user.
This needs to be refactored anyway since it's not necessary to to store this information in the database anymore, since I added the conversion from the cron String into `minute, hour, ...`
So the code in the if condition should always be executed to get those parameters.
But this needs some refactoring before it can be removed.